sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts

```diff
@@ -0,0 +1,109 @@
+import { createPipeline } from "./pipeline";
+import { convertSchemaIdResponse, convertSchemaResponse } from "./conversions";
+
+import {
+  GetSchemaByIdOptions,
+  GetSchemaIdOptions,
+  SchemaDescription,
+  SchemaIdResponse,
+  SchemaResponse,
+  SchemaRegistryClientOptions,
+  RegisterSchemaOptions
+} from "./models";
+
+/**
+ * Client for Azure Schema Registry service.
+ */
+export class SchemaRegistryClient {
+  /** The Schema Registry service endpoint URL. */
+  readonly endpoint: string;
+
+  /** Underlying autorest generated client. */
+  private readonly client: GeneratedSchemaRegistryClient;
+
+  /**
+   * Creates a new client for Azure Schema Registry service.
+   *
+   * @param endpoint The Schema Registry service endpoint URL, for example
+   *                 https://mynamespace.servicebus.windows.net.
+   * @param credential Credential to authenticate requests to the service.
+   * @param options Options to configure API requests to the service.
+   */
+  constructor(
+    endpoint: string,
+    credential: TokenCredential,
+    options: SchemaRegistryClientOptions = {}
+  ) {
+    const pipeline = createPipeline(options, credential);
+    this.endpoint = endpoint;
+    this.client = new GeneratedSchemaRegistryClient(endpoint, { ...pipeline, endpoint });
+  }
+
+  /**
+   * Registers a schema.
+   *
+   * If schema of specified name does not exist in the specified group, a schema
+   * is created at version 1. If schema of specified name exists already in
+   * specified group, schema is created at latest version + 1.
+   *
+   * @param schema Schema to register.
+   * @return Registered schema's ID.
+   */
+  async registerSchema(
+    schema: SchemaDescription,
+    options?: RegisterSchemaOptions
+  ): Promise<SchemaIdResponse> {
+    const response = await this.client.schema.register(
+      schema.group,
+      schema.name,
+      // cast due to https://github.com/Azure/autorest.typescript/issues/715
+      // serialization type is an extensible enum, and therefore any string
+      // should be allowed.
+      schema.serializationType as SerializationType,
+      schema.content,
+      options
+    );
+    return convertSchemaIdResponse(response);
+  }
+
+  /**
+   * Gets the identity of an existing schema with matching name, group, type,
+   * and content.
+   *
+   * @param schema Schema to match.
+   * @return Matched schema's ID.
+   */
+  async getSchemaId(
+    schema: SchemaDescription,
+    options?: GetSchemaIdOptions
+  ): Promise<SchemaIdResponse> {
+    const response = await this.client.schema.queryIdByContent(
+      schema.group,
+      schema.name,
+      // cast due to https://github.com/Azure/autorest.typescript/issues/715
+      // serialization type is an extensible enum, and therefore any string
+      // should be allowed.
+      schema.serializationType as SerializationType,
+      schema.content,
+      options
+    );
+    return convertSchemaIdResponse(response);
```

Comment:
Apologies if this is already discussed.. what is the reason for not using the response directly here?
Is it because it has extra properties that we don't want to expose or have properties using different names than what we want?

Reply:
The generated names are pretty bad right now with leading x from the x-header and xSchemaType is also about to be renamed.

With some work on the swagger it could be possible to get better names using http://azure.github.io/autorest/extensions/#x-ms-client-name. 

I mentioned this gently on the swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/10220#discussion_r469587082. That PR has gotten a ton of feedback and iteration, and I was rather late with that point and it wasn't blocking to just adapt the response like this.

After talking to @MiYanni, the plan is to go over his names and shape on Tuesday with an eye towards standardizing between languages. We should feed back the name choices down to the swagger too.

Reply:
> The generated names are pretty bad right now with leading x from the x-header. And xSchemaType is also about to be renamed to xSerializationType. I've used the good name already sans x prefix.

I believe we can use what are called "transforms" when running the code generator that will use the desired names in the generated code. Going over the shape with Yanni and Adam sounds like a good idea.

Reply:
I played with the transforms and so far I cannot find a way to get this to be named anything but body

https://github.com/Azure/azure-sdk-for-js/blob/7a26e1df850f86df774c263aa05baf51092aec7f/sdk/schemaregistry/schema-registry/src/generated/models/index.ts#L111

Swagger:
https://raw.githubusercontent.com/Azure/azure-rest-api-specs/a90b9146e543d3eec11381bd52d3b6ff271b1b78/specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

How I renamed the others:
https://gist.github.com/nguerrera/0621e7d58e69aabe948e037f5b5bb0db